mbuild/compound.py

```diff
@@ -1662,6 +1662,31 @@ def _visualize_nglview(self, show_ports=False, color_scheme={}):
         overwrite_nglview_default(widget)
         return widget
 
+    def flatten(self):
+        """Flatten the hierarchical structure of the Compound."""
+        # TO DO: handle ports when flatten out a structure
```

Comment:
```python
ports_list =list(self.all_ports()) #I think you want all ports here, not available ports but I may be mistaken.
# remove ports, I think this may get done in the remove method, but we would have to test it:
# https://github.com/mosdef-hub/mbuild/blob/a1611f1512ed348306cd90b844ba061b4324a5ed/mbuild/compound.py#L793
for port in ports_list():
    self.add(port, containment=False, label="*")
```